src/Components/Web/src/Forms/DisplayName.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Forms;
+
+/// <summary>
+/// Displays the display name for a specified field, reading from <see cref="DisplayAttribute"/>
+/// or <see cref="DisplayNameAttribute"/> if present, or falling back to the property name.
+/// </summary>
+/// <typeparam name="TValue">The type of the field.</typeparam>
+public class DisplayName<TValue> : ComponentBase
```

Comment:
Does this need to extend ComponentBase? Ideally we should be able to only re-render if the actual final string has changed